host/src/clocks.rs

```diff
@@ -0,0 +1,131 @@
+use crate::{wasi_clocks, wasi_default_clocks, WasiCtx};
+use anyhow::Context;
+
+pub struct MonotonicClock {
+    start: std::time::Instant,
+}
+
+impl Default for MonotonicClock {
+    fn default() -> MonotonicClock {
+        MonotonicClock {
+            start: std::time::Instant::now(),
+        }
+    }
+}
+
+impl MonotonicClock {
+    fn now(&self) -> std::time::Duration {
+        std::time::Instant::now().duration_since(self.start)
+    }
+    fn resolution(&self) -> std::time::Duration {
+        // FIXME bogus value
+        std::time::Duration::from_millis(1)
+    }
+    fn new_timer(&self, initial: std::time::Duration) -> MonotonicTimer {
+        MonotonicTimer { initial }
+    }
+}
+
+pub struct MonotonicTimer {
+    initial: std::time::Duration,
+}
+
+impl MonotonicTimer {
+    fn current(&self) -> std::time::Duration {
+        // FIXME totally bogus implementation
+        self.initial
+    }
+}
+
+#[derive(Default)]
+pub struct WallClock;
+
+impl WallClock {
+    fn now(&self) -> chrono::DateTime<chrono::Utc> {
+        chrono::Utc::now()
```

Comment:
No good reason, I think I googled for "rust datetime" and just went with chrono because it was the first type that came up, and I didn't remember what the deal with std's was.

Reply:
I got rid of it